in_toto/keylib.go

```diff
@@ -2,220 +2,402 @@ package in_toto
+				Public:  strings.TrimSpace(hex.EncodeToString(pubKeyBytes)),
+			}
+		} else {
+			k.KeyVal = KeyVal{
+				Public: strings.TrimSpace(hex.EncodeToString(pubKeyBytes)),
+			}
+		}
+	default:
+		return fmt.Errorf("%w: %s", ErrUnsupportedKeyType, keyType)
+	}
+	k.KeyType = keyType
+	k.Scheme = scheme
+	k.KeyIdHashAlgorithms = keyIdHashAlgorithms
+	if err := k.GenerateKeyId(); err != nil {
 		return err
 	}
+	return nil
+}
 
-	hashed := sha256.Sum256(data)
+/*
+ParseKey tries to parse a PEM []byte slice.
+Supported are:
 
-	// Create hex bytes from the signature hex string
-	sigHex, _ := hex.DecodeString(sig.Sig)
+	* PKCS8
+	* PKCS1
+	* PKIX
 
-	// SecSysLib uses a SaltLength of `hashes.SHA256().digest_size`, i.e. 32
-	if err := rsa.VerifyPSS(rsaPub, crypto.SHA256, hashed[:], sigHex,
-		&rsa.PSSOptions{SaltLength: sha256.Size, Hash: crypto.SHA256}); err != nil {
-		return err
+On success it returns the parsed key and nil.
+On failure it returns nil and the error ErrFailedPEMParsing
+*/
+func ParseKey(data []byte) (interface{}, error) {
+	key, err := x509.ParsePKCS8PrivateKey(data)
+	if err == nil {
+		return key, nil
 	}
-
-	return nil
+	key, err = x509.ParsePKCS1PrivateKey(data)
+	if err == nil {
+		return key, nil
+	}
+	key, err = x509.ParsePKIXPublicKey(data)
+	if err == nil {
+		return key, nil
+	}
+	return nil, ErrFailedPEMParsing
 }
 
 /*
-ParseEd25519FromPrivateJSON parses an ed25519 private key from the json string.
-These ed25519 keys have the format as generated using in-toto-keygen:
+LoadKey loads the key file at specified file path into the key object.
+It automatically derives the PEM type and the key type.
+Right now the following PEM types are supported:
 
-	{
-		"keytype: "ed25519",
-		"scheme": "ed25519",
-		"keyid": ...
-		"keyid_hash_algorithms": [...]
-		"keyval": {
-			"public": "..." # 32 bytes
-			"private": "..." # 32 bytes
-		}
-	}
-*/
-func ParseEd25519FromPrivateJSON(JSONString string) (Key, error) {
+	* PKCS1 for private keys
+	* PKCS8	for private keys
+	* PKIX for public keys
+
+The following key types are supported:
+
+	* ed25519
+	* RSA
+	* ecdsa
 
-	var keyObj Key
-	err := json.Unmarshal([]uint8(JSONString), &keyObj)
+On success it will return nil. The following errors can happen:
+
+	* path not found or not readable
+	* no PEM block in the loaded file
+	* no valid PKCS8/PKCS1 private key or PKIX public key
+	* errors while marshalling
+	* unsupported key types
+*/
+func (k *Key) LoadKey(path string, scheme string, keyIdHashAlgorithms []string) error {
+	pemFile, err := os.Open(path)
 	if err != nil {
-		return keyObj, fmt.Errorf("this is not a valid JSON key object")
+		return err
 	}
-
-	if keyObj.KeyType != "ed25519" || keyObj.Scheme != "ed25519" {
-		return keyObj, fmt.Errorf("this doesn't appear to be an ed25519 key")
+	defer func() {
+		if closeErr := pemFile.Close(); closeErr != nil {
+			err = closeErr
+		}
+	}()
+	// Read key bytes and decode PEM
```

Comment:
PEM decoding does not happen in ReadAll, right? 

```suggestion
	// Read key bytes
```